cmake/modules/CephChecks.cmake

```diff
@@ -70,7 +70,28 @@ check_symbol_exists(res_nquery "resolv.h" HAVE_RES_NQUERY)
 check_symbol_exists(F_SETPIPE_SZ "linux/fcntl.h" CEPH_HAVE_SETPIPE_SZ)
 check_symbol_exists(__func__ "" HAVE_FUNC)
 check_symbol_exists(__PRETTY_FUNCTION__ "" HAVE_PRETTY_FUNC)
-check_symbol_exists(getentropy "unistd.h" HAVE_GETENTROPY)
+
+include(CheckCSourceRuns)
+check_c_source_runs("
+  #include <errno.h>
+  #include <unistd.h>
+
+  int main() {
+    char buf[20];
+    int ret;
+    ret = getentropy(buf, 20);
+    if (ret < 0 && errno == ENOSYS) // Function not implemented
```

Comment:
@liuchang0812 i think this is a runtime behavior of this syscall. it will still fail if we happen to compile ceph on a system with glibc 2.25 + Linux 3.17, and end up running the compiled executable on a system with Linux < 3.17.

probably a better fix is to check it at runtime. see #30544